src/Compilers/CSharp/Portable/Symbols/ConstraintsHelper.cs

```diff
@@ -858,16 +858,30 @@ public static bool CheckMethodConstraints(
             ref ArrayBuilder<TypeParameterDiagnosticInfo> useSiteDiagnosticsBuilder,
             BitVector skipParameters = default(BitVector))
         {
-            return CheckConstraints(
-                method,
-                in args,
-                method.TypeSubstitution,
-                ((MethodSymbol)method.OriginalDefinition).TypeParameters,
-                method.TypeArgumentsWithAnnotations,
-                diagnosticsBuilder,
-                nullabilityDiagnosticsBuilderOpt,
-                ref useSiteDiagnosticsBuilder,
-                skipParameters);
+            bool constraintsSatisfied = true;
+            if (method.Arity > 0 && method.TypeSubstitution is not null)
+            {
+                constraintsSatisfied &= CheckConstraints(
+                     method,
+                     in args,
+                     method.TypeSubstitution,
+                     method.OriginalDefinition.TypeParameters,
+                     method.TypeArgumentsWithAnnotations,
+                     diagnosticsBuilder,
+                     nullabilityDiagnosticsBuilderOpt,
+                     ref useSiteDiagnosticsBuilder,
+                     skipParameters);
+            }
+
+            if (method.GetIsNewExtensionMember() && method.ContainingType is { Arity: > 0 } extension
```

Comment:
>if (method.GetIsNewExtensionMember() && method.ContainingType is { Arity: > 0 } extension [](http://example.com/codeflow?start=12&length=89)

As we discussed offline, I think this logic should be moved into the method called above, and quite possibly into much lower level helper, the "root". Therefore, I do not expect implementation of the enclosing method to change.